daemon/archive_tarcopyoptions_unix.go

```diff
@@ -14,12 +18,21 @@ func (daemon *Daemon) tarCopyOptions(container *container.Container, noOverwrite
 		return daemon.defaultTarCopyOptions(noOverwriteDirNonDir), nil
 	}
 
-	user, err := idtools.LookupUser(container.Config.User)
+	user, err := user.Lookup(container.Config.User)
```

Comment:
Resolving the in-container user name to a UID by looking it up in the host's user database looks highly suspect, but it's been this way since the original implementation in #28923.

Reply:
Yes; I recall now that I came to that conclusion as well; https://github.com/moby/moby/issues/34142 (also recall I worked on some things related to this package, but it's been a long time ago 😂 https://github.com/moby/moby/pull/34143)